### PR TITLE
KAFKA-17568 Rewrite TestPurgatoryPerformance by Java

### PR DESCRIPTION
**Pull request by m1a2st**
Jira: https://issues.apache.org/jira/browse/KAFKA-17568
Move TestPurgatoryPerformance from Scala to Java, and transfer to use benchmark.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by m1a2st** [comment]
| Benchmark                                          | (numKeys) | (numPossibleKeys) | (numRequests) | (pct50) | (pct75) | (requestRate) | (timeout) | Mode  | Cnt | Score              | Error           | Units |
|----------------------------------------------------|-----------|-------------------|---------------|---------|---------|---------------|-----------|-------|-----|--------------------|-----------------|-------|
| TestPurgatoryPerformance.testPurgatoryPerformance  | 3         | 100               | 100           | 0.5     | 0.75    | 10.0          | 1000      | thrpt | 15  | ≈ 10⁻¹⁰            |                 | ops/ns|
| TestPurgatoryPerformance.testPurgatoryPerformance  | 3         | 100               | 100           | 0.5     | 0.75    | 10.0          | 1000      | avgt  | 15  | 11568900851.367    | ± 725751170.308 | ns/op |

File: jmh-benchmarks/src/main/java/org/apache/kafka/jmh/core/TestPurgatoryPerformance.java [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+        purgatory.shutdown();
+    }
+
+    @Benchmark
+    public void testPurgatoryPerformance() throws InterruptedException {
+        LatencySamples latencySamples = new LatencySamples(1000000, pct75, pct50);
+        IntervalSamples intervalSamples = new IntervalSamples(1000000, requestRate);
+        CountDownLatch latch = new CountDownLatch(numRequests);
+        
+        List<String> keys = IntStream.range(0, numRequests)
```
Comment (chia7712) [inline jmh-benchmarks/src/main/java/org/apache/kafka/jmh/core/TestPurgatoryPerformance.java]: We should prepare those objects before running benchmark.

**Review by chia7712 — COMMENTED** [review]
@m1a2st could you please rebase code?

**Review by chia7712 — COMMENTED** [review]
@m1a2st the test pattern might not be easy to use with JMH. How about rewriting it in plain Java instead of using JMH?

**Comment by m1a2st** [comment]
Thanks for your suggestion, I will change it to plain Java

**Comment by m1a2st** [comment]
my local mashine can run the test
image

**Review by chia7712 — APPROVED** [review]

**Pull request merged**